packages/core/src/browser/tree/tree-widget.ts

```diff
@@ -117,17 +129,183 @@ export class TreeWidget extends VirtualWidget implements StatefulWidget {
+    }
+
+    protected decorateCaption(node: ITreeNode, attrs: ElementAttrs): ElementAttrs {
+        const style = this.getDecorationData(node, 'fontData').filter(notEmpty).reverse().map(fontData => this.applyFontStyles({}, fontData)).reduce((acc, current) =>
+            ({
+                ...acc,
+                ...current
+            })
+            , {});
+        return {
+            ...attrs,
+            style
+        };
+    }
+
+    protected applyFontStyles(original: ElementInlineStyle, fontData: FontData | undefined) {
+        if (fontData === undefined) {
+            return original;
+        }
+        let modified = original;
+        const { color, style } = fontData;
+        if (color) {
+            modified = {
+                ...modified,
+                color
+            };
+        }
+        if (style) {
+            (Array.isArray(style) ? style : [style]).forEach(s => {
+                switch (style) {
+                    case 'bold':
+                        modified = {
+                            ...modified,
+                            fontWeight: style
+                        };
+                        break;
+                    case 'normal': // Fall through.
+                    case 'oblique': // Fall through.
+                    case 'italic':
+                        modified = {
+                            ...modified,
+                            fontStyle: style
+                        };
+                        break;
+                    case 'underline': // Fall through.
+                    case 'line-through':
+                        modified = {
+                            ...modified,
+                            textDecoration: style
+                        };
+                        break;
+                    default:
+                        throw new Error(`Unexpected font style: ${style}.`);
+                }
+            });
+        }
+        return modified;
+    }
+
+    protected renderCaptionSuffixes(node: ITreeNode, props: NodeProps): h.Child[] {
+        return this.getDecorationData(node, 'captionSuffix').filter(notEmpty).reduce((acc, current) => acc.concat(current), []).map(suffix => {
+            const style = this.applyFontStyles({}, suffix.fontData);
+            const attrs = {
+                className: DecoratorStyles.CAPTION_SUFFIX,
+                style
+            };
+            return h.div(attrs, suffix.data);
+        });
+    }
+
+    protected renderCaptionPrefix(node: ITreeNode, props: NodeProps): h.Child {
+        const prefix = this.getDecorationData(node, 'captionPrefix').filter(notEmpty).shift();
+        if (prefix) {
+            const style = this.applyFontStyles({}, prefix.fontData);
+            const attrs = {
+                className: DecoratorStyles.CAPTION_PREFIX,
+                style
+            };
+            return h.div(attrs, prefix.data);
+        }
+        // tslint:disable-next-line:no-null-keyword
```

Comment:
just fyi: you can declare it once at the top: `// tslint:disable:no-null-keyword`